src/scalar.rs

```diff
@@ -415,6 +525,42 @@ impl Scalar {
     }
 
     /// Compute the multiplicative inverse of this scalar.
+    ///
+    /// # Warning
+    ///
+    /// All input `Scalars` **MUST** be nonzero.  If you cannot
```

Comment:
This should just say that `self` must be nonzero (since unlike the batch case we just have one scalar).